addons/crm/models/crm_lead.py

```diff
@@ -1303,12 +1303,26 @@ def _merge_opportunity(self, user_id=False, team_id=False, auto_unlink=True, max
         return opportunities_head
 
     def _merge_get_fields_specific(self):
-        return {
+        fields_specific = {
             'description': lambda fname, leads: '<br/><br/>'.join(desc for desc in leads.mapped('description') if not is_html_empty(desc)),
             'type': lambda fname, leads: 'opportunity' if any(lead.type == 'opportunity' for lead in leads) else 'lead',
             'priority': lambda fname, leads: max(leads.mapped('priority')) if leads else False,
+            'tag_ids': lambda fname, leads: leads.mapped('tag_ids'),
+            'lost_reason': lambda fname, leads:
+                False if leads and leads[0].probability
+                else next((lead.lost_reason for lead in leads if lead.lost_reason), False),
         }
 
+        # Address related fields, all fields are propagated or none
+        fields_specific.update({
```

Comment:
Looking at it, it seems it is not really all fields or none, more "first found containing any field". Shouldn't we try to get a bit more information  (like country on first lead, then full address on second lead) ?

Reply:
I did your idea bellow :) (address with the most non-empty fields)